### ISSUE TITLE
Significant Delay When Launching App or Switching Scenarios After Update to Version 3.0.2

### ISSUE THREAD

**Issue by AlmightyCZ** [Nain57/Smart-AutoClicker#482]
### Version of Smart AutoClicker

3.0.2

### Device brand and model

Samsung S23

### Version of your Android OS

14

### Type

Smart scenario

### Describe what happened?

After updating from version 2.4.2 to 3.0.2, I noticed a significant delay (approximately 3 seconds, but it feels much longer) when launching the app or when returning to the app through a notification to change a scenario. I have 15 smart scenarios set up. I was able to temporarily eliminate the delay by reinstalling the app and reimporting all the scenarios, but this fix only lasted a few hours. The delay has since returned to the same level as it was after the update.

Here are some videos demonstrating the behavior:

https://github.com/user-attachments/assets/09c907ec-c53e-4668-96fe-1e1e74379b0c

https://github.com/user-attachments/assets/034690ca-2dc9-48d3-9e8c-590cb076ed47

### Describe what should have happened?

On 2.4.2 everything was smooth and instant.

**Comment by Nain57** [comment]
Hi and thank you for your feedback.

One clarification, did you buy the app on the PlayStore ?

Because after analysis, I have found a problem in the code than can reproduce your issue when the app is bought. Basically, once bought, I stop the advertisement SDK and thus, no more data is collected so I also stop the user consent SDK. But when the app was starting, it was checking the wrong purchase state and was waiting for the maximum time for a user consent SDK answer, which is never happening.

tldr; if you are using the PlayStore version and have bought the app, this should be fixed by https://github.com/Nain57/Smart-AutoClicker/commit/2e734811c73bc5f3b58525ef3cf521eeb0f4a13a

**Comment by AlmightyCZ** [comment]
Thank you for resolving the issue. I’ve been satisfied with the app and purchased it early as it met my expectations. I’m looking forward to the next release with the fix. Appreciate the work you've put into it.

**Comment by Nain57** [comment]
Closed, release will be available once built (~30mins) here: https://github.com/Nain57/Smart-AutoClicker/releases/tag/3.0.4
If you can still reproduce the issue, feel free to open it again
